provider/status_update_template_resource.go

```diff
@@ -0,0 +1,128 @@
+			"body": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"id": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func readResourceFireHydrantStatusUpdateTemplate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	firehydrantAPIClient := m.(firehydrant.Client)
+
+	id := d.Id()
+	tflog.Debug(ctx, fmt.Sprintf("Read status update template: %s", id), map[string]interface{}{
+		"id": id,
+	})
+
+	template, err := firehydrantAPIClient.StatusUpdateTemplates().Get(ctx, id)
+	if err != nil {
+		if errors.Is(err, firehydrant.ErrorNotFound) {
+			tflog.Debug(ctx, fmt.Sprintf("Status update template %s does not exist", id), map[string]interface{}{
+				"id": id,
+			})
+			d.SetId("")
+			return nil
+		}
+		return diag.Errorf("Error reading status update template %s: %v", id, err)
+	}
+
+	attributes := map[string]interface{}{
+		"name": template.Name,
+		"body": template.Body,
+		"id":   template.ID,
+	}
+	for key, value := range attributes {
+		if err := d.Set(key, value); err != nil {
+			return diag.Errorf("Error setting %s for status update template %s: %v", key, id, err)
+		}
+	}
+
+	return diag.Diagnostics{}
+}
+
+func createResourceFireHydrantStatusUpdateTemplate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	firehydrantAPIClient := m.(firehydrant.Client)
+
+	createRequest := firehydrant.CreateStatusUpdateTemplateRequest{
+		Name: d.Get("name").(string),
+		Body: d.Get("body").(string),
+	}
+
+	tflog.Debug(ctx, fmt.Sprintf("Create status update template: %s", d.Id()), map[string]interface{}{
+		"id": d.Id(),
+	})
+	statusUpdateTemplateResponse, err := firehydrantAPIClient.StatusUpdateTemplates().Create(ctx, createRequest)
+	if err != nil {
+		return diag.Errorf("Error creating status update template %s: %v", d.Id(), err)
+	}
+
+	d.SetId(statusUpdateTemplateResponse.ID)
+
+	return readResourceFireHydrantStatusUpdateTemplate(ctx, d, m)
+}
+
+func updateResourceFireHydrantStatusUpdateTemplate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	firehydrantAPIClient := m.(firehydrant.Client)
+
+	updateRequest := firehydrant.UpdateStatusUpdateTemplateRequest{
+		Name: d.Get("name").(string),
+		Body: d.Get("body").(string),
+	}
+
+	tflog.Debug(ctx, fmt.Sprintf("Update status update template: %s", d.Id()), map[string]interface{}{
+		"id": d.Id(),
+	})
+	_, err := firehydrantAPIClient.StatusUpdateTemplates().Update(ctx, d.Id(), updateRequest)
+	if err != nil {
+		return diag.Errorf("Error updating status update template %s: %v", d.Id(), err)
+	}
+
+	return readResourceFireHydrantStatusUpdateTemplate(ctx, d, m)
+}
+
+func deleteResourceFireHydrantStatusUpdateTemplate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	firehydrantAPIClient := m.(firehydrant.Client)
+
+	tflog.Debug(ctx, fmt.Sprintf("Delete status update template: %s", d.Id()), map[string]interface{}{
```

Comment:
```suggestion
	tflog.Debug(ctx, "Delete status update template", map[string]interface{}{
```
Seems like we can remove some redundancy? Same with all the `tflog` lines above.

Reply:
i usually leave error message static and put the variables in the following key-value arg, which ends up being printed as structured logging i think? the `map[string]interface{}` will print out `id=123` from what i remember

Reply:
Ok.  Works for me.  Fixing.